streams/src/main/java/org/apache/kafka/streams/processor/internals/Task.java

```diff
@@ -26,26 +26,64 @@
 import java.util.Set;
 
 public interface Task {
+    enum State {
+        CREATED, RESTORING, RUNNING, SUSPENDED, CLOSED;
+
+        static void validateTransition(final State oldState, final State newState) {
```

Comment:
`Task.java:32:9: Cyclomatic Complexity is 18 (max allowed is 16). [CyclomaticComplexity]`

Reply:
A later commit fixes the cyclomatic complexity, but the NPath complexity is too high. I'll add a suppression for it.